### PR TITLE
fix: snap avatar colors

### PR DESCRIPTION
**Pull request by FrederikBolding**
## **Description**

Adjusts the background and border color of `SnapAvatar` by request of @eriknson. I had to use `style` since the color we wanna use is not available in `BackgroundColor`.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24346?quickstart=1)

### CODE REVIEW

File: ui/components/app/snaps/snap-avatar/snap-avatar.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ const SnapAvatar = ({
     >
       {iconUrl ? (
         <AvatarFavicon
-          backgroundColor={BackgroundColor.backgroundAlternative}
+          style={{ "background-color": 'var(--color-background-alternative-hover)' }}
```
Comment (Mrtenz) [inline ui/components/app/snaps/snap-avatar/snap-avatar.js:67]: Why is this necessary?

Comment (FrederikBolding) [reply ui/components/app/snaps/snap-avatar/snap-avatar.js:67]: The color we wanted to use is not available in `BackgroundColor` or on the components we use 😞

Comment (Mrtenz) [reply ui/components/app/snaps/snap-avatar/snap-avatar.js:67]: Can we add it to `BackgroundColor`?

Comment (FrederikBolding) [reply ui/components/app/snaps/snap-avatar/snap-avatar.js:67]: Yeah, but it doesn't work since there is no generated CSS class for `Box` to use this color.

**Review by david0xd — DISMISSED** [review]

**Review by david0xd — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]
